ublox_serialization/include/ublox_serialization/serialization.hpp

```diff
@@ -386,6 +389,8 @@ class Reader {
       if (data_[0] == options_.sync_a &&
           (count_ == 1 || data_[1] == options_.sync_b)) {
         break;
+      } else {
+        extra_data_.push_back(data_[0]);
```

Comment:
Thinking about this more, I'm a little concerned that this is eventually going to use up all memory in the system.  That is, we never end up clearing out `extra_data_`, so it is just going to grow indefinitely as we `push_back` data.  I think that probably during `getExtraData`, we should return a *copy* of this data, and then clear out `extra_data_`.  That will ensure we don't "leak" memory.

Does that make sense to you?

Reply:
I don't _think_ this is an issue (it looks like the `Reader` only exists for the duration of the `readCallback`, so the memory should be freed every time), but I figured better safe than sorry and implemented anyway.